src/libraries/Microsoft.Extensions.DependencyInjection.Abstractions/src/ActivatorUtilities.cs

```diff
@@ -291,11 +291,11 @@ static void InitializeCtorArgValues(ref object[] ctorArgs, int length)
 #endif
             CreateFactoryInternal(instanceType, argumentTypes, out ParameterExpression provider, out ParameterExpression argumentArray, out Expression factoryExpressionBody);
 
-            var factoryLambda = Expression.Lambda<Func<IServiceProvider, object?[]?, object>>(
+            var factoryLambda = Expression.Lambda<ObjectFactory>(
                 factoryExpressionBody, provider, argumentArray);
 
-            Func<IServiceProvider, object?[]?, object>? result = factoryLambda.Compile();
-            return result.Invoke;
+            ObjectFactory? result = factoryLambda.Compile();
```

Comment:
Note for other reviewers: ObjectFactory delegate is declared here: https://github.com/dotnet/runtime/blob/da757a1b062c24e1011c5aadf31fe2f1cde59c97/src/libraries/Microsoft.Extensions.DependencyInjection.Abstractions/src/ObjectFactory.cs#L14